rest/src/main/kotlin/request/RequestHandler.kt

```diff
@@ -15,5 +20,17 @@ public interface RequestHandler {
      * @throws RestRequestException when a non-rate limit error response is returned.
      */
     @Throws(RestRequestException::class)
-    public suspend fun <B : Any, R> handle(request: Request<B, R>): R
+    public suspend fun <B : Any, R> handle(request: Request<B, R>, context: Any? = null): R
+
+    public suspend fun <T> intercept(builder: RequestBuilder<T>, context: Any? = null): RequestBuilder<T> {
+        return builder
+    }
+}
+
+public fun  <T> RequestBuilder<T>.defaultInterception(route: Route<T>, token: String): RequestBuilder<T> {
+        if (route.requiresAuthorizationHeader) {
+            unencodedHeader(HttpHeaders.UserAgent, KordConstants.USER_AGENT)
+            unencodedHeader(HttpHeaders.Authorization, "Bot $token")
+        }
+    return this
```

Comment:
```suggestion
    public suspend fun <T> intercept(builder: RequestBuilder<T>, context: Any? = null): RequestBuilder<T> =
        builder.defaultInterception(token)
}

public fun  <T> RequestBuilder<T>.defaultInterception(token: String): RequestBuilder<T> = apply {
    unencodedHeader(HttpHeaders.UserAgent, KordConstants.USER_AGENT)
    if (route.requiresAuthorizationHeader) {
        unencodedHeader(HttpHeaders.Authorization, "Bot $token")
    }
```
default interception in `RequestHandler.intercept` default implementation, always provide user agent, use `route` from `RequestBuilder`